File: app/src/main/java/audiolibros/example/com/audiolibros/Aplicacion.java
package audiolibros.example.com.audiolibros;

import android.app.Application;
import android.graphics.Bitmap;
import android.support.v4.util.LruCache;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.ImageLoader;
import com.android.volley.toolbox.Volley;

import java.util.List;

/**
 * Created by Miguel Á. Núñez on 29/01/2018.
 */

public class Aplicacion extends Application {
    private List<Libro> listaLibros;
    private AdaptadorLibrosFiltro adaptador;
    private static RequestQueue colaPeticiones;
    private static ImageLoader lectorImagenes;

    @Override
    public void onCreate() {
        super.onCreate();
        listaLibros = Libro.Companion.ejemploLibros();
        adaptador = new AdaptadorLibrosFiltro(this, listaLibros);
        colaPeticiones = Volley.newRequestQueue(this);
        lectorImagenes = new ImageLoader(colaPeticiones, new ImageLoader.ImageCache() {
            private final LruCache<String, Bitmap> cache = new LruCache<>(10);

            public void putBitmap(String url, Bitmap bitmap) {
                cache.put(url, bitmap);
            }

            public Bitmap getBitmap(String url) {
                return cache.get(url);
            }
        });
    }

    public AdaptadorLibrosFiltro getAdaptador() {
        return adaptador;
    }

    public List<Libro> getListaLibros() {
        return listaLibros;
    }

    public static RequestQueue getColaPeticiones() {
        return colaPeticiones;
    }

    public static ImageLoader getLectorImagenes() {
        return lectorImagenes;
    }
}

File: app/src/main/java/audiolibros/example/com/audiolibros/Aplicacion.kt
package audiolibros.example.com.audiolibros

import android.app.Application
import android.graphics.Bitmap
import android.support.v4.util.LruCache

import com.android.volley.RequestQueue
import com.android.volley.toolbox.ImageLoader
import com.android.volley.toolbox.Volley

/**
 * Created by Miguel Á. Núñez on 29/01/2018.
 */

class Aplicacion : Application() {
    var listaLibros: MutableList<Libro>? = null
    var adaptador: AdaptadorLibrosFiltro? = null

    override fun onCreate() {
        super.onCreate()
        listaLibros = Libro.ejemploLibros()
        adaptador = AdaptadorLibrosFiltro(this, listaLibros!!)
        colaPeticiones = Volley.newRequestQueue(this)
        lectorImagenes = ImageLoader(colaPeticiones, object : ImageLoader.ImageCache {
            private val cache = LruCache<String, Bitmap>(10)

            override fun putBitmap(url: String, bitmap: Bitmap) {
                cache.put(url, bitmap)
            }

            override fun getBitmap(url: String): Bitmap? {
                return cache.get(url)
            }
        })
    }

    fun getLectorImagenes(): ImageLoader? {
        return lectorImagenes
    }


    companion object {
        var colaPeticiones: RequestQueue? = null
            private set
        var lectorImagenes: ImageLoader? = null
            private set
    }
}
